grpc-client-server/README.md

```diff
@@ -0,0 +1,181 @@
+# GRPC Client and Server Communication 
+
+The goals of this project are: 
+ * to use gRPC requests, and understand how they differ from REST
+ * to use deadlines with gRPCs
+ * to build a simple prober
+ * optionally, gain experience with instrumenting an application and running Prometheus
+
+## What is gRPC?
+
+RPCs aren’t the same as HTTP requests. HTTP is a text-based protocol, oriented towards resources or entities. RPCs are oriented towards calling a specific function. RPCs are highly structured; normally the request and response are encoded in efficient binary formats which are not human-readable. HTTP APIs are usually limited to CRUD (Create, Read, Update, Delete) operations and RPCs can perform any kind of operation. If you want to efficiently integrate two systems that you control, RPCs are a good choice. If you want to provide an API for use by developers outside of your organisation then HTTP APIs are generally a better choice, because they are simpler to develop against and all programming languages provide good HTTP support. gRPC, which we will use in this exercise, is a RPC implementation that is fairly common in industry. 
+
+Read the [gRPC Introduction](https://grpc.io/docs/what-is-grpc/introduction/) and 
+[gRPC Core Concepts](https://grpc.io/docs/what-is-grpc/core-concepts/) for an overview of gRPC.
+todo stuff from primer
+
+## Get started with a HelloWorld gRPC example
+
+Start by building a simple gRPC ‘hello world’ server and cli client in golang. See [gRPC Quickstart](https://grpc.io/docs/languages/go/quickstart/) for detailed instructions.
+
+## Change HelloWorld to Prober
```

Comment:
so i can always lay out a sample project to get them here but I quite like teaching the implicit skill of find an example chunk of code and hack on it... your call, tom

Reply:
I think the "don't do this in real life" instruction further makes me feel less good about this. Is there a way to teach it that _doesn't_ require renaming?